### PR TITLE
DOTORG-849: Add "Create Constituent Action" to Blackbaud RENXT

### PR DESCRIPTION
**Pull request by noahcooper**
This PR updates the Blackbaud Raiser's Edge NXT Destination to add the "Create Constituent Action" Action.
Similar to the "Create Gift" Action, if a constituent ID is not provided, a new constituent can be created if other constituent fields are provided. 

Example payload:

```
{
  "date": "2023-03-03T21:35:02.794Z",
  "category": "Task/Other",
  "completed": true,
  "completed_date": "2023-03-03T21:35:02.794Z",
  "description": "Signed Petition",
  "direction": "Inbound",
  "fundraisers": [
    "192"
  ],
  "priority": "Normal",
  "summary": "Signed Petition",
  "constituent_id": "625960"
}
```

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
hi @noahcooper acknowledging this PR. Will get it reviewed ASAP.

File: packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createConstituentAction/index.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  },
+  start_time: {
+    label: 'Start Time',
+    description: 'The start time of the action. Uses 24-hour time in the HH:mm format. For example, 17:30 represents 5:30 p.m.',
+    type: 'string'
+  },
+  status: {
+    label: 'Status',
+    description:
+      'The action status. If the system is configured to use custom action statuses, available values are the entries in the Action Status table.',
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createConstituentAction/index.ts:111]: Is this description something Raisers Edge users will understand? What happens if a value is sent which doesn't match an entry in the table?

**Review by joe-ayoub-segment — COMMENTED** [review]
Generally looks good to me @noahcooper . 
Couple of minor comments only, which probably amounts to nothing! 

Local tests pass.

File: packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createConstituentAction/index.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  },
+  fundraisers: {
+    label: 'Fundraisers',
+    description: 'The set of immutable constituent system record IDs for the fundraisers associated with the action.',
+    type: 'string',
+    multiple: true
+  },
+  location: {
+    label: 'Location',
+    description: 'The location of the action. Available values are the entries in the Action Locations table.',
```
[the remark this answers is not in the document]
Comment (noahcooper) [reply packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createConstituentAction/index.ts:75]: Yes, this is something Raiser's Edge admins should understand. Tested and confirmed that providing an invalid value results in a 400 response:

```
{
  "category": "Task/Other",
  "constituent_id": "625960",
  "location": "foo"
}
```

Returns a 400 with the message:

```
The table 'Action Locations' does not contain an active entry named 'foo'
```

File: packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createConstituentAction/index.ts (hunk 1 above)
Comment (noahcooper) [reply packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createConstituentAction/index.ts:111]: Yes, this is something Raiser's Edge admins should understand. Tested and confirmed that providing an invalid value results in a 400 response:

```
{
  "category": "Task/Other",
  "constituent_id": "625960",
  "status": "foo"
}
```

Returns a 400 with the message:

```
The table 'Action Status' does not contain an active entry named 'foo'
```

**Review by joe-ayoub-segment — APPROVED** [review]
Looks good. Minor comments which @noahcooper clarified. 
This change adds a new Action and refactors code for another Action. 
The Integration is in Private Beta and has no customers using it yet.

**Comment by joe-ayoub-segment** [comment]
hi @noahcooper this PR has been deployed.

**Pull request merged**